### ISSUE TITLE
Long-press option to hide Search

### ISSUE THREAD

**Issue by dgcampea** [falzonv/discreet-launcher#91]
Being a [Arcticons](https://github.com/Donnnno/Arcticons) user, I'd like to be able to swap the 'Search' icon for consistency reasons.
Currently it doesn't look like Turtl can pick this icon up to submit to the icon pack author.

**Comment by falzonv** [comment]
Hello,

> it doesn't look like Turtl can pick this icon up

Indeed, it is probably because the Search is not a real app. It is added dynamically when creating the app list and is displayed via a "PopupWindow" (not even an "Activity" so there is no entry point like a classic Android app).

*For information, the vector icon for Search is generated from [app/src/main/res/drawable/icon_search.xml](https://github.com/falzonv/discreet-launcher/blob/main/app/src/main/res/drawable/icon_search.xml).*

Unfortunately I already tried, for something else, to do an icon picker from an icon pack and I was not satisfied with the result (from what I recall it was quite complex to implement and could consume lots of resources if the pack contains many icons, there was also issue to save the selected icon, but clean it if the user uninstall the icon pack, etc.).

---

Currently, I have two other ideas that could solve the issue:

1/ When an icon pack is selected and if it contains the Discreet Launcher icon, automatically reuse this icon for the Search (otherwise, use the default Search icon).

or

2/ Although it would work only with Arcticons, we could make a special statement to recognize a specially created Discreet Launcher Search icon. The line to add in the "appfilter.xml" of Arcticons would be something like the following (drawable name to adapt) and I would also have to make some changes in the Discreet Launcher code to manage it.

```
<item component="ComponentInfo{discreetlauncher.search/discreetlauncher.search}" drawable="discreetlaunchersearch" />
```

---

@tuch96: if I remember correctly you are also using Arcticons.
What do you think about this? (current Search icon and the two proposals above)

Best regards.

**Comment by falzonv** [comment]
Hello @dgcampea,

Something I just thought about: do you actually use the Search?
If not, an alternative (and much simpler) solution would be to hide its icon (with the "Hidden apps" menu like for other apps).

Best regards.

**Comment by tuch96** [comment]
Hi Vincent, the solution to hide the icon over "hidden apps" is the best solution for that point. It works with Arcticons!

**Comment by dgcampea** [comment]
> Something I just thought about: do you actually use the Search?
> If not, an alternative (and much simpler) solution would be to hide its icon (with the "Hidden apps" menu like for other apps).

Personally I don't use it, hiding it would be fine by me.

**Comment by dgcampea** [comment]
Something like _Long press_ on the icon to show a "Hide this app" option? Right now holding it or pressing it does the same action.
I think the confusion comes from the fact that "Search" isn't really an app (that people install) but a feature provided by the Launcher.

**Comment by falzonv** [comment]
Very good idea, thank you!
I will see how to do this with the long press, let's reuse this ticket for follow-up :-)

*Remark: I will be away from Internet and GitHub for a couple of weeks.*

**Comment by falzonv** [comment]
Hello,

I am pleased to inform you that this has been implemented in new release v5.0.0!
This release has been published few minutes ago.

Here are the usual availability delays for each location:
- 4 to 8 days on F-Droid
- 0 to 2 days on Google Play
- Already available as direct APK on GitHub (https://github.com/falzonv/discreet-launcher/releases)

*Remarks:
I will leave this ticket opened until the release is effectively published on F-Droid.
You can use the export/import feature to switch from one version to another.*

Best regards.

**Comment by falzonv** [comment]
Since v5.0.0 is now available on F-Droid, I will close this ticket.
Feel free to add comments if needed or wished.
